src/EventStore.Client.Streams/StreamSubscription.cs

```diff
@@ -63,34 +64,52 @@ public class StreamSubscription : IDisposable {
 		private async Task Subscribe() {
 			_log.LogDebug("Subscription {subscriptionId} confirmed.", SubscriptionId);
 			using var _ = _disposed;
+
 			try {
 				await foreach (var resolvedEvent in _events.ConfigureAwait(false)) {
 					try {
-						_log.LogTrace("Subscription {subscriptionId} received event {streamName}@{streamRevision} {position}",
-							SubscriptionId, resolvedEvent.OriginalEvent.EventStreamId,
-							resolvedEvent.OriginalEvent.EventNumber, resolvedEvent.OriginalEvent.Position);
+						_log.LogTrace(
+							"Subscription {subscriptionId} received event {streamName}@{streamRevision} {position}",
+							SubscriptionId,
+							resolvedEvent.OriginalEvent.EventStreamId,
+							resolvedEvent.OriginalEvent.EventNumber,
+							resolvedEvent.OriginalEvent.Position
+						);
 						await _eventAppeared(this, resolvedEvent, _disposed.Token).ConfigureAwait(false);
-					} catch (Exception ex) when (ex is ObjectDisposedException or OperationCanceledException) {
+					} catch (Exception ex) when
+						(ex is ObjectDisposedException or OperationCanceledException) {
 						if (_subscriptionDroppedInvoked != 0) {
 							return;
 						}
-						_log.LogWarning(ex,
+
+						_log.LogWarning(
+							ex,
 							"Subscription {subscriptionId} was dropped because cancellation was requested by another caller.",
-							SubscriptionId);
+							SubscriptionId
+						);
 
 						SubscriptionDropped(SubscriptionDroppedReason.Disposed);
 
 						return;
 					} catch (Exception ex) {
-						_log.LogError(ex,
+						_log.LogError(
+							ex,
 							"Subscription {subscriptionId} was dropped because the subscriber made an error.",
-							SubscriptionId);
+							SubscriptionId
+						);
 						SubscriptionDropped(SubscriptionDroppedReason.SubscriberError, ex);
 
 						return;
 					}
 				}
-			} catch (Exception ex) {
+			} catch (RpcException ex) when (ex.Status.StatusCode == StatusCode.Cancelled &&
+			                                ex.Status.Detail.Contains("Call canceled by the client.")) {
```

Comment:
At the moment this hard coded string is checked in tests for `grpc-dotnet`; this is safe, for now.

Reply:
Maybe checking the status code is enough. I am not sure if the `Cancelled` status code produces any other message.